File: programming/Euler/Euler2.java
import java.util.*;
public class Euler2{
    public static void main(String[] args){
	int a=0;
	int b=1;
	ArrayList<Integer> list=new ArrayList<Integer>();
	while(b<4000000){
	    a=b+a;
	    b=a+b;
	    if(a%2==0) list.add(a);
	    if(b%2==0) list.add(b);

	}
	System.out.println(list.stream().mapToDouble(c->c).sum());
    }
}

File: programming/Euler/Euler2/src/Euler2.kt
fun main(){
    var a=3
    var b=5
    println(a+b)
    var sum=10
    while (true){
        val a2=a+b+b
        val b2=a2+a+b
        a=a2
        b=b2
        println(a2+b2)
        if(a+b>4000000) break;
        sum+=(a2+b2)
    }
    println(sum);
}